src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/OffenderRepository.java

```diff
@@ -21,17 +20,19 @@
 @Repository
 public interface OffenderRepository extends JpaRepository<Offender, Long> {
     List<Offender> findByNomsId(String nomsId);
+
     List<Offender> findByLastNameAndFirstNameAndBirthDate(final String lastName, final String firstName, final LocalDate dob);
 
     @Query("select o from Offender o left join fetch o.bookings b WHERE o.nomsId = :nomsId order by b.bookingSequence asc")
     List<Offender> findOffendersByNomsId(@Param("nomsId") String nomsId, Pageable pageable);
 
-    default Optional<Offender> findOffenderByNomsId(String nomsId){
-        return findOffendersByNomsId(nomsId, PageRequest.of(0,1)).stream().findFirst();
+    default Optional<Offender> findOffenderByNomsId(String nomsId) {
+        return findOffendersByNomsId(nomsId, PageRequest.of(0, 1)).stream().findFirst();
     }
 
-    @EntityGraph(value = "offender-with-sentences")
-    Optional<Offender> findFirstWithSentencesByNomsId(String nomsId);
+    @Query("select o from Offender o left join fetch o.bookings b left join fetch b.releaseDetail r " +
+        "WHERE o.nomsId = :nomsId order by b.bookingSequence asc limit 1")
+    Optional<Offender> findOffendersWithReleaseDetailByNomsId(@Param("nomsId") String nomsId);
```

Comment:
Really struggled with this one - the entity graph was returning too much information, so had to revert to just grabbing the details using native sq.  The limit 1 is just to mimic the previous call that was using .findFirst();(line 30).
This however does show _HHH90003004: firstResult/maxResults specified with collection fetch; applying in memory_ (as does the findFirst method)

Reply:
So essentially you want the latest bookingId? So I suspect you want to to say 
"select o from Offender o left join fetch o.bookings b left join fetch b.releaseDetail r  where o.nomisId = :nomisId and b.bookingSequence = 1"

Would that always return a single row? Maybe try it in SQL first

Reply:
ah ok - sounds reasonable - didn't think about that option... will try